index.js

```diff
@@ -28,230 +25,277 @@ const IMPORT_PATH_DEFAULT = {
+      CallExpression: {
+        exit(p) {
+          if (!universalImportSpecifier
+            || p.node.callee.name !== universalImportSpecifier) {
+            return
+          }
+          const [universalArgPath] = p.get('arguments')
+          const { blocking, path } = findImport(universalArgPath, t, imports)
+          if (!path) return
+          transformImport(path, t, { blocking, templates, options: this.opts })
+        }
+      },
       Import(p) {
-        if (p[visited]) return
-        p[visited] = true
-
-        const importArgNode = getImportArgPath(p).node
-        t.existingChunkName = existingMagicCommentChunkName(importArgNode)
-        // no existing chunkname, no problem - we will reuse that for fixing nested chunk names
-
-        const universalImport = getImport(p, IMPORT_UNIVERSAL_DEFAULT)
-
-        // if being used in an await statement, return load() promise
-        if (
-          p.parentPath.parentPath.isYieldExpression() || // await transformed already
-          t.isAwaitExpression(p.parentPath.parentPath.node) // await not transformed already
-        ) {
-          const func = t.callExpression(universalImport, [
-            loadOption(t, loadTemplate, p, importArgNode).value,
-            t.booleanLiteral(false)
-          ])
-
-          p.parentPath.replaceWith(func)
+        if (!universalImportSpecifier) return
+        const importCall = p.parentPath
+        if (importCall.parentPath.isVariableDeclarator()) {
+          imports[importCall.parentPath.node.id.name] = { blocking: false, path: p }
           return
         }
-
-        const opts = (this.opts.babelServer
-          ? [
-            idOption(t, importArgNode),
-            this.opts.includeFileName ? fileOption(t, p) : undefined,
-            pathOption(t, pathTemplate, p, importArgNode),
-            resolveOption(t, resolveTemplate, importArgNode),
-            chunkNameOption(t, chunkNameTemplate, importArgNode)
-          ]
-          : [
-            idOption(t, importArgNode),
-            this.opts.includeFileName ? fileOption(t, p) : undefined,
-            loadOption(t, loadTemplate, p, importArgNode), // only when not on a babel-server
-            pathOption(t, pathTemplate, p, importArgNode),
-            resolveOption(t, resolveTemplate, importArgNode),
-            chunkNameOption(t, chunkNameTemplate, importArgNode)
-          ]
-        ).filter(Boolean)
-
-        const options = t.objectExpression(opts)
-
-        const func = t.callExpression(universalImport, [options])
-        delete t.existingChunkName
-        p.parentPath.replaceWith(func)
+        const blocking = isBlockingExpression(importCall.parentPath)
+        if (isReturnStatement(importCall)
+          || (blocking && isReturnStatement(importCall.parentPath))) {
+          const fnDeclaration = getFunctionDeclaration(importCall.parentPath)
+          if (fnDeclaration) {
+            imports[fnDeclaration.node.id.name] = { blocking, path: p }
```

Comment:
```suggestion
          const declaration = getFunctionDeclaration(importCall.parentPath)
          if (declaration) {
            imports[declaration.node.id.name] = { blocking, path: p }
```

It's pretty much explicit that a "function" declaration is returned.